Fixed handling of html captions on gallery and image.

jorgefilipecosta opened the pull request:
When adding an image with existing HTML captions they were not being parsed. Now we use the existing HTML parsing mechanisms (hpqParse plus children matcher) to correctly parse the captions.

Fixes: https://github.com/WordPress/gutenberg/issues/5338

## How Has This Been Tested?
Follow instructions provided in https://github.com/WordPress/gutenberg/issues/5338 and verify the problem is now fixed and we correctly use the HTML that was in the caption instead of showing the

iseulde commented on blocks/library/gallery/block.js:
```diff
@@ -86,7 +88,7 @@ class GalleryBlock extends Component {
 			images: images.map( ( attributes ) => ( {
 				...attributes,
-				caption: attributes.caption ? [ attributes.caption ] : [],
+				caption: hpqParse( attributes.caption, children() ),
```
I feel like this should be part of the attributes declaration, rather than something afterwards. Would love to have @aduth's opinion/suggestion.

iseulde replied:
hm also how does this belong in `onSelectImages`? When is it not being parsed exactly?

iseulde replied:
I think whatever calls `onSelectImages` on the image(s) should do the parsing?

jorgefilipecosta replied:
Hi @iseulde when we select an image from the mediaLibrary the media library pass us a string with a caption. The mediaLibrary should not handle the creation of the react children as the user of the library may, in fact, want things in string format. In order for us to correctly pass that html and use it in our captions RichText fields we need to pass it to the "react children format". That's exactly what this line is doing.

gziolo replied:
I think that you can do it inside the `MediaUpload` component as @iseulde suggested. You would have only one place where this transformation happens and all blocks would benefit from it.

https://github.com/WordPress/gutenberg/blob/a2f41b7206a87ce56995b2a869fa2701a095866c/blocks/media-upload/index.js#L113-L117

jorgefilipecosta replied:
Yes @iseulde and @gziolo I think it may make sense to do it in MediaUpload because it is part of the blocks, I had the idea it was part of the components and so it would not be good to include things from the block parsing. I will update the code.

jorgefilipecosta replied:
Code was updated.

jorgefilipecosta committed:
Fixed handling of html captions on gallery and image.

When adding an image with existing html captions they were not being parsed. Now we use the existing html parsing mechanisms (hpqParse plus children matcher) to correctly parse the captions.

gziolo reviewed (approved):
Great job, it works like a charm! 💯
